core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -169,6 +169,16 @@
    AuthorizationManager getAuthorizationManager();
 
    /**
+    * Whenever this cache acquires a lock it will do so using the given Object as the owner of said lock.
+    * <p>
+    * This method provides no guarantees as to how this affect locking and should not be used for general usage.
```

Comment:
Yeah tbh this is not really designed for users to use. But I can try to be more specific as to how it works or I could be even more forceful in saying this shouldn't be used by an end user (only says not for general usage currently).